src/peer_connection/operation/mod.rs

```diff
@@ -53,9 +53,20 @@ impl Operations {
     /// the execution will start immediately in a new goroutine.
     pub(crate) async fn enqueue(&self, op: Operation) -> Result<()> {
         if let Some(ops_tx) = &self.ops_tx {
-            let _ = ops_tx.send(op)?;
-            self.length.fetch_add(1, Ordering::SeqCst);
+            return Operations::enqueue_inner(op, &ops_tx, &self.length);
         }
+
+        Ok(())
+    }
+
+    fn enqueue_inner(
+        op: Operation,
+        ops_tx: &Arc<mpsc::UnboundedSender<Operation>>,
+        length: &Arc<AtomicUsize>,
+    ) -> Result<()> {
+        length.fetch_add(1, Ordering::SeqCst);
```

Comment:
I flipped the order of sending on the channel and increment the length too. Feel safer to use this ordering.